cmd/create_test.go

```diff
@@ -33,9 +28,7 @@ func TestCreate_Execute(t *testing.T) {
 func TestCreate_NoRuntime(t *testing.T) {
 	defer fromTempDir(t)()
 
-	// command with a client factory which yields a fully default client.
-	cmd := NewCreateCmd(func(ClientOptions) *fn.Client { return fn.New() })
-	cmd.SetArgs([]string{})
```

Comment:
Args are only passed through if the `--` delimiter is used, which is the user of the test package explicitly asking to pass flags through to the test, which is useful for test debugging without needing to make a code change.   Just normal args will not interfere (test flags).

Reply:
>Just normal args will not interfere (test flags)

Really? In my experience I had issues when executing specific test using `go test -run "REGEXHERE"`.

Reply:
For instance if I remove the same thing from `TestDelete_ByProject` then test fails:
```
=== RUN   TestDelete_ByProject
    delete_test.go:76: expected name 'bar', got '^\QTestDelete_ByProject\E$'
```

Reply:
I believe is really should be explicitly set to empty slice.

Reply:
@lkingland ^^^

Reply:
But maybe it's given by how tests are run in Intellij Idea.

Reply:
Idea first compiles tests and then it runs it, and that's when issues happens.

Reply:
```
GOROOT=/home/mvasek/go17 #gosetup
GOPATH=/home/mvasek/go #gosetup
/home/mvasek/go17/bin/go test -c -o /tmp/GoLand/___TestDelete_ByProject_in_knative_dev_kn_plugin_func_cmd.test knative.dev/kn-plugin-func/cmd #gosetup
/home/mvasek/go17/bin/go tool test2json -t /tmp/GoLand/___TestDelete_ByProject_in_knative_dev_kn_plugin_func_cmd.test -test.v -test.paniconexit0 -test.run ^\QTestDelete_ByProject\E$
```

Reply:
Interesting; this is not my experience running `go test` directly.  Maybe it is indeed an IDE difference.  I will run a few more tests locally to get more information.

Reply:
I think you are right that this might be an issue with the IDE or environment.  I was able to confirm locally that only arguments after the `--` delimiter are actually passed through to the code being tested.  Also , first compiling the tests before running them does not recreate the issue for me:
```
# cd cmd
# go test -c
# go test -test.run=TestDelete_ByProject
PASS
```

Reply:
Quick update, this actually fails when not using the `=`.  This is due to a nuance of Cobra, which uses its own flag parsing package.  To avoid this in the future I did add back the []string{} args, _and_ added it to all other commands which do not explicitly set args.